### PR TITLE
[in-process] 1st stage: compilation cl->bc

### PR DESCRIPTION
**Pull request by emankov**
To switch on in-process compilation Compiler::SetInProcess() should be called on the invoker's side.

### CODE REVIEW

File: src/driver/AmdCompiler.cpp [hunk 1]
```diff
@@ -522,15 +535,62 @@ bool AMDGPUCompiler::CompileToLLVMBitcode(Data* input, Data* output, const std::
...
 
   for (const std::string& s : options) {
     args.push_back(s.c_str());
   }
 
-  bool res = InvokeDriver(args);
-  if (res) { output->ReadOutputFile(bcFile); }
-  return res;
+  if (inprocess) {
+    std::unique_ptr<Driver> driver(new Driver(llvmBin + "/clang", "amdgcn-amd-amdhsa-opencl", diags));
```
Comment (rampitec) [inline src/driver/AmdCompiler.cpp]: Eventually we will need to run without actual llvmBin and compiler binaries. Can we still compiler without a real clang executable argument? Maybe just pass "clang" or ""?

Comment (emankov) [inline src/driver/AmdCompiler.cpp]: Driver's constructor doesn't check the executable existence, thus empty string might be passed, I double checked it.

Comment (rampitec) [inline src/driver/AmdCompiler.cpp]: Let's then remove artificial dependency on the llvmBin.

Comment (emankov) [inline src/driver/AmdCompiler.cpp]: done

**Comment by nhaustov** [comment]
Instead of boolean flag, could it be a separate implementation class AMDGPUInProcessCompiler?

**Comment by nhaustov** [comment]
Note: you can also copy the file completely, rename the class and then start replacing the parts with in-process code.

**Pull request merged**